### PR TITLE
feat(agent): Add Local Forwarder support for Agent

### PR DESCRIPTION
**Pull request by aroberts87**
## What this PR does / why we need it:
* Create a new configmap for the local forwarder
* Add local forwarder elements to json schema
* Add volume mapping for new configmap

## Checklist

- [x] Title of the PR starts with type and scope, (e.g. feat(agent,node-analyzer,sysdig-deploy):)
- [x] Chart Version bumped for the respective charts
- [x] Variables are documented in the README.md (or README.tpl in some charts)
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [x] All test files are added in the tests folder of their respective chart and have a "_test" suffix

### CODE REVIEW

File: charts/agent/templates/_helpers.tpl [hunk 1]
```diff
@@ -506,3 +506,8 @@ sysdig_capture_enabled: false
 true
 {{- end }}
 {{- end }}
+
+{{/* Return the name of the local forwarder configmap */}}
+{{- define "agent.localForwarderConfigMapName" }}
+{{- include "agent.configmapName" . | printf "%s-local-forwarder" | trunc 63 | trimSuffix "-" }}
```
Comment (mavimo) [inline charts/agent/templates/_helpers.tpl]: If the agent configmapName is loger we will cut out the `local-forwarder` suffix creating clash, I suggest to cut the previous part and then append the suffix, like:
```suggestion
{{- include "agent.configmapName" . | trunc 46 | trimSuffix "-" | printf "%s-local-forwarder" }}
```

File: charts/agent/values.schema.json [hunk 2]
```diff
@@ -23,6 +24,100 @@
...
       }
+    },
+    "localForwarder": {
+      "type": "object",
+      "required": [
+        "enabled",
+        "integrations",
+        "transmitMessageTypes"
+      ],
+      "if": {
```
Comment (mavimo) [inline charts/agent/values.schema.json:35]: FYI: Not sure if the "if" here is supported on old version of `helm` CLI

File: charts/agent/values.schema.json [hunk 3]
```diff
@@ -23,6 +24,100 @@
...
+                "SECURE_AUDIT"
+              ]
+            }
+          }
+        }
+      }
+    }
+  },
+  "$defs": {
+    "integration": {
```
Comment (mavimo) [inline charts/agent/values.schema.json]: let's use a less ambiguous name here ;) 
```suggestion
    "AgentLocalForwarderIntegration": {
```

**Review by mavimo — CHANGES_REQUESTED** [review]

File: charts/agent/values.yaml [hunk 4]
```diff
@@ -280,6 +280,13 @@ tolerations:
     value: "true"
 leaderelection:
   enable: false
+localForwarder:
+  enabled: false
+  transmitMessageTypes:
+    - AUDIT_TAP
```
Comment (aroberts87) [inline charts/agent/values.yaml]: This needs to be removed.

**Comment by claffin** [comment]
Can we include these in the readme as well? @aroberts87

**Comment by aroberts87** [comment]
> Can we include these in the readme as well? @aroberts87

Absolutely, @claffin. I'm reviving this PR now and will include the entries in the readme.

File: charts/agent/values.schema.json (hunk 2 above)
Comment (aroberts87) [reply charts/agent/values.schema.json:35]: It appears to be...

I've been able to successfully use this change with helm 3.6.3 and 3.10.1. I can look into some other approaches here if you'd like to remove the if/then flow though.

File: charts/agent/values.schema.json (hunk 3 above)
Comment (aroberts87) [reply charts/agent/values.schema.json]: Handled in 372a28b

**Review by mavimo — APPROVED** [review]

**Pull request merged**